do_cmake.sh

```diff
@@ -1,8 +1,12 @@
 #!/usr/bin/env bash
 set -x
+
 git submodule update --init --recursive
-if test -e build; then
-    echo 'build dir already exists; rm -rf build and re-run'
+
+[ -z "$BUILD_DIR" ] && BUILD_DIR=build
+
+if test -e $BUILD_DIR; then
```

Comment:
we could be more consistent when using test or `[ foobar ]` ?

Reply:
@tchaikov why "should" when we can have "must" ([shellcheck](https://github.com/koalaman/shellcheck), which btw enforces `[` over `test`)? It's available in quite a few [distros](https://repology.org/project/shellcheck/versions)